ndb/tests/unit/test_async_.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+
+from google.cloud.ndb import _eventloop
+from google.cloud.ndb import async_
+
+
+@unittest.mock.patch("google.cloud.ndb.async_._eventloop.EventLoop")
```

Comment:
Why not patch the actual module `"google.cloud.ndb._eventloop.EventLoop"`?

Reply:
https://docs.python.org/3/library/unittest.mock.html#id5

Reply:
Are you saying the test failed when patching `google.cloud.ndb._eventloop`. `mock` will "do the right thing" when patching a **module** because `sys.modules` is globally available. This would be different if you had done

```python
from google.cloud.ndb._eventloop import EventLoop
```

----

Example showing that module patching works "correctly":

```python
# a.py
import b


def f():
    return b.g() + 10
```

```python
# b.py
def g():
    return 88
```

```python
# test_a.py
import unittest.mock

import a


def test_f():
    assert a.f() == 98


@unittest.mock.patch("b.g", return_value=1)
def test_f_patch_b(b_g):
    assert a.f() == 11
    b_g.assert_called_once_with()


@unittest.mock.patch("a.b.g", return_value=10)
def test_f_patch_a(a_b_g):
    assert a.f() == 20
    a_b_g.assert_called_once_with()
```

Reply:
Oh, I see. Yeah, I did that weird.  I don't remember how I arrived at that.

Reply:
No worries. That pattern is right when individual members are imported, but luckily we seldom / never import anything but modules.